windows/src/engine/keyman32/kmprocess.cpp

```diff
@@ -451,24 +451,30 @@ BOOL ProcessGroup(LPGROUP gp)
 	*/
 
 	p = kkp->dpOutput;
-	if(*p != UC_SENTINEL || *(p+1) != CODE_CONTEXT)
-	{
-		for(p = _td->miniContext; *p; p = incxstr(p))
-		{
-			if(*p == UC_SENTINEL)
-				switch(*(p+1))
-				{
-			    case CODE_DEADKEY: _td->app->QueueAction(QIT_BACK, BK_DEADKEY); break;
-					case CODE_NUL: break;	// 11 Aug 2003 - I25(v6) - mcdurdin - CODE_NUL context support
+	if(*p != UC_SENTINEL || *(p+1) != CODE_CONTEXT) {
+		for(PWSTR mcp = decxstr(wcschr(_td->miniContext, 0), _td->miniContext); mcp != NULL; mcp = decxstr(mcp, _td->miniContext)) {
```

Comment:
We need to add these in reverse order so we get the correlation with deadkey and surrogate pair elements.